r-package/grf/R/input_utilities.R

```diff
@@ -81,8 +81,8 @@ validate_clusters <- function(clusters, X) {
     clusters <- vector(mode="numeric", length=0)
   } else if (length(clusters) == 0) {
     clusters <- vector(mode="numeric", length=0)
-  } else if (!is.vector(clusters) | !all(clusters == floor(clusters))) {
-    stop("Clusters must be a vector of integers.")
+  } else if (!is.vector(as.numeric(clusters)) | !all(as.numeric(clusters) == floor(as.numeric(clusters)))) {
```

Comment:
This might be cleaner if we have two separate code paths for factors and non-factors... Something like if clusters is a factor, then just return the levels of the factor shifted by -1, else do the thing we did before. Currently this is rather hard to read.